From a08cf5a575325435040d35710dbac3d0c32b1676 Mon Sep 17 00:00:00 2001
From: Jan Beich <jbeich@FreeBSD.org>
Date: Fri, 15 Sep 2023 16:04:34 +0200
Subject: [PATCH] Drop unused symbols from version script after 5263c249495b

LLD 16 enabled --no-undefined-version by default but temporarily[1]
downgraded errors to warnings. LLD 17 now fails to build:

ld: error: version script assignment of 'global' to symbol 'sd_bus_attach_event' failed: symbol not defined
ld: error: version script assignment of 'global' to symbol 'sd_bus_detach_event' failed: symbol not defined
ld: error: version script assignment of 'global' to symbol 'sd_bus_get_event' failed: symbol not defined

[1] https://github.com/llvm/llvm-project/commit/c1949c6a310ae965be4995f46e9b47c56dc5f05f
---
 src/libsystemd/libsystemd.sym | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/libsystemd/libsystemd.sym b/src/libsystemd/libsystemd.sym
index cdb775a..de32b5e 100644
--- a/src/libsystemd/libsystemd.sym
+++ b/src/libsystemd/libsystemd.sym
@@ -72,9 +72,6 @@ global:
         sd_bus_get_current_message;
         sd_bus_get_current_handler;
         sd_bus_get_current_userdata;
-        sd_bus_attach_event;
-        sd_bus_detach_event;
-        sd_bus_get_event;
         sd_bus_add_filter;
         sd_bus_add_match;
         sd_bus_add_object;
-- 
2.38.5

